### PR TITLE
Add sysinfo and update LLVM

### PR DESCRIPTION
**Pull request by VocalFan**
Due to us seeming to find OS specific and Hardware specific issues more and more, I decided to add sysinfo to display the Operating System, CPU, and Available/Total RAM. I also updated LLVM from 17.0.2 to 17.0.5 while also adding it where we hash by URL instead of file like Ultima suggested.
```
Operating System    : Windows 11 Pro 22621
CPU Information     : AMD Ryzen 9 3900X 12-Core Processor            
Memory Available    : 53100/65461 MB
```

### CODE REVIEW

**Review by ultimaweapon — CHANGES_REQUESTED** [review]
You need to separated cache key creation from download step and skip the download step if hash of the URL is already cached.

**Comment by VocalFan** [comment]
Yeah I noticed that too, but a migraine made me go to sleep while I waited for the llvm url to build xD

**Comment by VocalFan** [comment]
Yeah, got it working lol

**Comment by VocalFan** [comment]
MacOS moment

File: .github/workflows/ci-windows.yml [hunk 1]
```diff
@@ -10,16 +10,13 @@ jobs:
...
-        $hash = (Get-FileHash llvm.tar.xz).Hash
-        unxz llvm.tar.xz
-        tar -x --strip-components=1 -f llvm.tar
-        echo "hash=$hash" >> $env:GITHUB_OUTPUT
+        $url = "https://github.com/llvm/llvm-project/releases/download/llvmorg-17.0.5/llvm-project-17.0.5.src.tar.xz"
+        echo "llvmurl=$url" >> $env:GITHUB_OUTPUT
       id: download
     - name: Generate cache keys
-      run: echo "dist=${{ runner.os }}-llvm-${{ steps.download.outputs.hash }}" >> $env:GITHUB_OUTPUT
+      run: echo "dist=${{ runner.os }}-llvm-${{ steps.download.outputs.llvmurl }}" >> $env:GITHUB_OUTPUT
```
Comment (ultimaweapon) [inline .github/workflows/ci-windows.yml:19]: Better to hash the URL instead of using the URL itself so the key will not be too long.

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-windows.yml (hunk 1 above)
Comment (VocalFan) [reply .github/workflows/ci-windows.yml:19]: Not very pretty doing a hash through powershell xS

**Review by ultimaweapon — APPROVED** [review]

**Pull request merged**